src/components/pdf-upload/PdfUploadAgent.tsx

```diff
@@ -126,12 +166,21 @@ export const PdfUploadAgent = ({
         },
         body: JSON.stringify({
           sessionId,
-          fileName: file.name,
+          fileName: filename,
           fileSize: file.size,
```

Comment:
to my point about `description`, you're not even passing it here lolz. i'll stop talking about it, just merge, i'll delete in subsequent pr :)